File: DesignPatternsInAutomatedTesting-Series/PatternsInAutomation.Tests/Beginners/Pages/BingMainPage/BingMainPageValidator.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;

namespace PatternsInAutomatedTests.Beginners.Pages.BingMainPage
{
    public class BingMainPageValidator
    {
        private readonly IWebDriver browser;

        public BingMainPageValidator(IWebDriver browser)
        {
            this.browser = browser;
        }

        protected BingMainPageElementMap Map
        {
            get
            {
                return new BingMainPageElementMap(this.browser);
            }
        }

        public void ResultsCount(string expectedCount)
        {
            Assert.IsTrue(this.Map.ResultsCountDiv.Text.Contains(expectedCount), "The results DIV doesn't contains the specified text.");
        }
    }
}

File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/com/automatetheplanet/_1_page_object_pattern/Pages/BingMainPage/BingMainPageValidator.java
package com.automatetheplanet._1_page_object_pattern.Pages.BingMainPage;

import org.openqa.selenium.WebDriver;
import org.testng.Assert;

public class BingMainPageValidator {

    private final WebDriver _browser;

    public BingMainPageValidator(WebDriver browser) {
        _browser = browser;
    }

    protected BingMainPageElementMap map() {
        return new BingMainPageElementMap(_browser);
    }

    public void resultsCount(String expectedCount) {
        Assert.assertTrue(map().resultsCountDiv().getText().contains(expectedCount), "The results DIV doesn't contains the specified text.");
    }
}
